src/components/encoding-pane/range-filter-shelf.tsx

```diff
@@ -4,84 +4,100 @@ import * as React from 'react';
+export class RangeFilterShelfBase extends React.PureComponent<RangeFilterShelfProps, RangeFilterShelfState> {
+
+  constructor(props: RangeFilterShelfProps) {
     super(props);
     this.state = ({
       minDateTimePickerOpen: false,
       maxDateTimePickerOpen: false
     });
-
-    this.filterModifyMinBound = this.filterModifyMinBound.bind(this);
-    this.filterModifyMaxBound = this.filterModifyMaxBound.bind(this);
     this.filterModifyExtent = this.filterModifyExtent.bind(this);
+    this.filterModifyMaxBound = this.filterModifyMaxBound.bind(this);
+    this.filterModifyMinBound = this.filterModifyMinBound.bind(this);
     this.toggleMinDateTimePicker = this.toggleMinDateTimePicker.bind(this);
     this.toggleMaxDateTimePicker = this.toggleMaxDateTimePicker.bind(this);
   }
 
   public render() {
-    const {domain, filter, type} = this.props;
-    const range = filter.range;
-    const lowerBound = Math.floor(Number(domain[0]));
-    const upperBound = Math.ceil(Number(domain[1]));
-    let currMin, currMax, picker;
+    const {filter, domain, type} = this.props;
+    const createSliderWithTooltip = Slider.createSliderWithTooltip;
+    const Range = createSliderWithTooltip(Slider.Range);
+    let minInput, maxInput, formatLabel, currMin, currMax, lowerBound, upperBound;
     if (type === ExpandedType.TEMPORAL) {
-      currMin = new Date(range[0]);
-      currMax = new Date(range[1]);
-      picker = (
-        <div>
-          {this.renderDateTimePicker(currMin, 'min')}
-          {this.renderDateTimePicker(currMax, 'max')}
-        </div>
-      );
+      minInput = this.renderDateTimePicker(new Date(convertToTimestamp(filter.range[0] as DateTime)), 'min');
+      maxInput = this.renderDateTimePicker(new Date(convertToTimestamp(filter.range[1] as DateTime)), 'max');
+      currMin = convertToTimestamp(filter.range[0] as DateTime);
+      currMax = convertToTimestamp(filter.range[1] as DateTime);
+      // lowerBound = Math.floor(convertToTimestamp(domain[0] as DateTime));
+      // upperBound = Math.ceil(convertToTimestamp(domain[1] as DateTime));
+      formatLabel = this.formatTime;
     } else {
-      currMin = range[0];
-      currMax = range[1];
-      picker = (
-        <div>
-          {this.renderQuantitativeInput(Number(currMin), 'min')}
-          {this.renderQuantitativeInput(Number(currMax), 'max')}
-        </div>
-      );
+      minInput = this.renderNumberInput('min');
+      maxInput = this.renderNumberInput('max');
+      currMin = filter.range[0];
+      currMax = filter.range[1];
+    }
+    lowerBound = Math.floor(Number(domain[0]));
+    upperBound = Math.ceil(Number(domain[1]));
+    if (isDateTime(domain[0])) {
+      lowerBound = Math.floor(convertToTimestamp(domain[0] as DateTime));
+    }
+    if (isDateTime(domain[1])) {
+      upperBound = Math.ceil(convertToTimestamp(domain[1] as DateTime));
+    }
+    let step;
+    if (filter.timeUnit === TimeUnit.YEARMONTHDATE) {
```

Comment:
What about other timeUnit?  

Let's extract a method for any timeUnit specific rules and we can define step for all supported timeUnit and throw error for ones that we have not supported yet?  (This way when we want to add more, we can easily know we need to add it here.  

(Same for other things that you do specifically for timeUnits that we currently support.)